modules/sample/src/Module.php

```diff
@@ -25,13 +28,56 @@ public function init()
         parent::init();
         self::$instance = $this;
 
+        $this->_bindEvents();
+
         if (Craft::$app->request->getIsSiteRequest()) {
             Craft::$app->view->registerTwigExtension(new Extension());
         }
 
+        // Always turn on the debug bar in dev environment
+        if (
+            getenv('ENVIRONMENT') === 'dev' &&
+            !Craft::$app->getRequest()->getIsConsoleRequest() &&
+            !Craft::$app->getRequest()->getIsAjax()
+        ) {
+            Craft::$app->session->set('enableDebugToolbarForSite', true);
+        }
+
         Craft::info(
             'Sample module loaded',
             __METHOD__
         );
     }
+
+    private function _bindEvents()
+    {
+        // Add edit entry link to front-end templates
+        Event::on(
+            View::class,
+            View::EVENT_END_BODY,
+            function(Event $e) {
+                $element = Craft::$app->getUrlManager()->getMatchedElement();
+                
+                if (!$element) return;
```

Comment:
interesting what does this protect against? / what's the in-template equivalent of this check?

Reply:
This makes sure that you are on a URL that references a Craft element.

Reply:
Oh like for a route 👍